app/controllers/api/v2/providers_controller.rb

```diff
@@ -50,12 +50,13 @@ def update
 
       def suggest
         authorize Provider
-
         return render(status: :bad_request) if params[:query].nil? || params[:query].length < 3
         return render(status: :bad_request) unless begins_with_alphanumeric(params[:query])
 
+        normalise_query = CGI.unescape(params[:query]).downcase.gsub(/[^0-9a-z]/i, "")
```

Comment:
I'm surprised we have to do this, considering https://guides.rubyonrails.org/action_controller_overview.html#hash-and-array-parameters says that Rails already does this:

> The actual URL in this example will be encoded as "/clients?ids%5b%5d=1&ids%5b%5d=2&ids%5b%5d=3" as the "[" and "]" characters are not allowed in URLs. Most of the time you don't have to worry about this because the browser will encode it for you, and Rails will decode it automatically, but if you ever find yourself having to send those requests to the server manually you should keep this in mind.

Have we checked that unescaping is necessary?

Reply:
Also, I think that the normalisation should be happening in the `Provider` model, while the parameter, if any, should happen here in the controller. Sorry if that wasn't clear with what I said earlier, but since the normalisation column is in the model I think it belongs there, no?

Reply:
I'm not sure what you mean here? I had it in the Provider model scope but we discussed this yesterday and thought we wanted to move this out to the controller?

As for unescaping, it is necessary if the request being made has been escaped, it basically decodes it to the original state so we can normalize it ourselves

Reply:
What was suggested is that unescaping should be moved here, not the normalisation.

Reply:
re: escaping, so what you're saying is that we're escaping the escaping? i.e. we're doubling up on the escaping?

Reply:
#1324 splits this out into its own service if you look at the specs, you will see we have still used `CGI.unescape`. You are correct that Rack will be unescaping it for us but I still think it's worth leaving in here should this service be used for whatever reason and the query is not unescaped.